amplipi/streams.py

```diff
@@ -685,6 +704,11 @@ def send_cmd(self, cmd):
     except Exception as e:
       raise Exception(f"Error sending command {cmd}: {e}") from e
 
+  def validate_stream(self, **kwargs):
+    regex = r"^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$"
```

Comment:
I'm having a hard time digesting this regex. This is the one that detects a hostname-like name right?

Reply:
once this regex is cleaned up a potential better name for the `regex` variable might be `HOSTNAME_LIKE`